sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ExtRefService.java

```diff
@@ -130,4 +131,96 @@ private static SclReport createDataSetAndControlBlocks(SclRootAdapter sclRootAda
+        return extRefsByServiceType.keySet().stream()
+                .map(tServiceType-> {
+                    Set<TExtRef> tExtRefSet = extRefsByServiceType.get(tServiceType);
+                    return switch (tServiceType) {
+                        case REPORT -> checkLimitationForOneControlType(iedAdapter, tClientServices, tExtRefSet, CLIENT_IED_NAME+clientIedName+" subscribes to much REPORT Control Blocks.", ServicesConfigEnum.REPORT);
+                        case GOOSE -> checkLimitationForOneControlType(iedAdapter, tClientServices, tExtRefSet, CLIENT_IED_NAME+clientIedName+" subscribes to much GOOSE Control Blocks.", ServicesConfigEnum.GSE);
+                        case SMV -> checkLimitationForOneControlType(iedAdapter, tClientServices, tExtRefSet, CLIENT_IED_NAME+clientIedName+" subscribes to much SMV Control Blocks.", ServicesConfigEnum.SMV);
+                        default -> throw new IllegalStateException("Unsupported value: " + tServiceType);
+                    };
+                }).toList().stream()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .toList();
+    }
+
+    /**
+     * Checks Control Block number limitation for binded IED
+     * @param tExtRefs list of ExtRefs referenced same ied
+     * @param msg message to display hen error occured
+     * @param servicesConfigEnum type of Control Block for which check is done
+     * @return Optional of encountered error or empty
+     */
+    private Optional<SclReportItem> checkLimitationForOneControlType(IEDAdapter iedAdapter, TClientServices tClientServices, Set<TExtRef> tExtRefs, String msg, ServicesConfigEnum servicesConfigEnum) {
+        long max = getMaxInstanceAutorizedForBindedIED(tClientServices, servicesConfigEnum);
+        long value = tExtRefs.size();
+        return max == AccessPointAdapter.MAX_OCCURRENCE_NO_LIMIT_VALUE || value <= max ? Optional.empty() : Optional.of(SclReportItem.warning(iedAdapter.getXPath(), msg));
+    }
+
+    private Class<? extends TControl> getControlTypeClass(TServiceType tServiceType) {
+        return switch (tServiceType) {
+            case POLL -> null;
+            case REPORT -> TReportControl.class;
+            case GOOSE -> TGSEControl.class;
+            case SMV -> TSampledValueControl.class;
+        };
+    }
+
+    /**
+     * Gets max number authorized in configuration of each element DataSets, FCDAs, Control Blocks) into an AccessPoint
+     * @param servicesConfigEnum element type
+     * @return max number authorized by config
+     */
+    private long getMaxInstanceAutorizedForBindedIED(TClientServices tClientServices, ServicesConfigEnum servicesConfigEnum){
+
+        return switch (servicesConfigEnum) {
+            case DATASET, FCDA -> tClientServices.isSetMaxAttributes() ? tClientServices.getMaxAttributes() : AccessPointAdapter.MAX_OCCURRENCE_NO_LIMIT_VALUE;
```

Comment:
Is the switch's `case DATASET, FCDA` statement necessary? 
This situation won't ever be tested, making it useless.